apex-checks/src/main/java/org/fundacionjala/enforce/sonarqube/apex/checks/AssertMethodCheck.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.fundacionjala.enforce.sonarqube.apex.checks;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.Grammar;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+import org.sonar.squidbridge.checks.SquidCheck;
+
+import org.fundacionjala.enforce.sonarqube.apex.api.ApexKeyword;
+import org.fundacionjala.enforce.sonarqube.apex.api.grammar.RuleKey;
+
+/**
+ * Verifies if a test method contains invalid asserts.
+ */
+@Rule(
+        key = AssertMethodCheck.CHECK_KEY,
+        priority = Priority.MINOR,
+        name = "Invalid asserts in test methods",
+        description = "It's bad practice to use incorrectly assert methods",
+        tags = Tags.CONVENTION
+)
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
+@SqaleConstantRemediation("6min")
+@ActivatedByDefault
+public class AssertMethodCheck extends SquidCheck<Grammar> {
+
+    /**
+     * Stores an error message when to use 'assert(value, value)'.
+     */
+    public static final String ASSERT_EQUALS_MESSAGE
+            = "It's bad practice to use assert(value, value).";
+
+    /**
+     * Stores an error message when to use 'assert(true)'.
+     */
+    public static final String ASSERT_MESSAGE
+            = "It's bad practice to use assert(true).";
+
+    /**
+     * It is the code of the rule for the plugin.
+     */
+    public static final String CHECK_KEY = "A1008";
+
+    /**
+     * The variables are initialized and subscribe the base rule.
+     */
+    @Override
+    public void init() {
+        subscribeTo(RuleKey.METHOD_DECLARATION);
+    }
+
+    /**
+     * It is responsible for verifying whether the rule is met in the rule base.
+     * In the event that the rule is not correct, create message error.
+     *
+     * @param astNode It is the node that stores all the rules.
+     */
+    @Override
+    public void visitNode(AstNode astNode) {
+        if (!isTest(astNode)) {
+            return;
+        }
+        List<AstNode> expressions = astNode.getDescendants(RuleKey.ARGUMENTS);
```

Comment:
I think we can improve this part, we don't need to look for all ARGUMENTS nodes, just nodes related to assert, in this way we can reduce the amount of nodes to evaluate